### ISSUE TITLE
Option to sort by natural order of acquisition

### ISSUE THREAD

**Issue by ganqqwerty** [donkuri/Kaishi#2]
The Kaishi deck is sorted by frequency list. I propose the sorting by natural order of acquisition instead, because this way the learner will learn the words much faster with less stress.

### What's the problem with the current sorting? 
1. Words that are easier to remember, often referred to as 'low-hanging fruits', are not placed at the beginning of the deck. Instead, they are distributed throughout.
2.. There is a ton of words that occur often in the langauge but bear too abstract meaning. I mean stuff like する, ある, それ, あれ, そこ, おそこ, また, いつ and so on. Question words, particles, helper words and super abstract verbs, words that change horribly depending on conjugation. It's much harder to remember the word "how" or "have" or "who" than simple and concrete things like "mother", "happy/sad", "tasty/disgusting", "jump", "run", "apple", "brother", "face", "leg", "head", "bird", "dog", "good", "bad", etc. 

As a result, learners often feel overwhelmed by these abstract (or just hard-to-remember) words between days 2 and 15, leading to the abandonment of many pre-made decks.

#### Proposed solution

1) Locate a ordered vocabulary list that aligns with Krashen's theory of the natural order of acquisition. While Krashen focused on the natural order of grammar acquisition, there is research on vocabulary acquisition as well. For instance, [this paper](https://euacademic.org/UploadArticle/52.pdf) discusses vocabulary acquisition in children, and [this book](https://www.degruyter.com/document/doi/10.21832/9781847692900/html?lang=en) outlines general principles that apply to adult second language acquisition.
2) If no such list exists, manually reorder the words, distributing the more abstract words throughout the deck, removing some entirely, or placing them at the very end.

#### Some other optimizations
1. Antonyms should be placed close to each other or in direct succession.
2. Words within the same domain could follow one another, though this requires careful consideration.
3. Homonyms and words with similar sounds should be kept apart to prevent confusion if learned in a close timeframe.
4. Ideally, there should be a balance between short and long words to avoid sequences of many monosyllabic words.
5. Maybe we can think about which words do TMW newcomers specifically may know before starting learning Japanese. TMW attracts anime and pop-culture fans, can we suppose that a lot of them watched/heard of Naruto, for example, and know the word 影 as in Hokage or Kagebunchinnojutsu? Should the word 影 be placed earlier because of that? 

P.S. I know that the issue seems half-baked, because I didn't provide a concrete list. I'm ready to help with that, find the good research and re-order the deck.

**Comment by ganqqwerty** [comment]
The two sorting criteria can co-exist, too. We can have dedicated fields for different sorting criteria: the "frequency" field and the "natural order of acquisition" field. Similar thing was done in Jo Mako's Kanji deck, where he added RTK order, KKLC order, RRTK order, Wanikani order, etc. 

The anki user will be able to change the preferred order in the deck settings.

**Comment by ExecuteOrd66** [comment]
To add to this idea of reordering words according to ease of understanding. I recommend putting all ○然 words nearer to one another. This helps acquisition of the specific kanji.

In general my experience has been that it is beneficial to learn words that share kanji in close succession to one another (unless the kanji in these words have different readings).

**Comment by donkuri** [comment]
I like the idea, enhancing the deck is always welcome but I must admit I am not sure how to go about it. If you are willing to locate a list that makes sense and is backed by experimental data (or at the very least make one), I will consider it. The leading principle was having frequency sorting + manual sorting for the first 200 words. Tyogin took care of that though, so I do not know how exactly he went about it. I believe he mostly used i+1 guiding principles to order the first 200.

TL;DR Yeah that's a cool idea, but unless you come with something concrete I will not put a lot of time and effort into this for now.

**Comment by ganqqwerty** [comment]
I can come up with a concrete list, yes. What would be the best technical way to go about it? I guess a google spreadsheet with the new column for the alternative order will work fine?

**Comment by ganqqwerty** [comment]
Regarding the experimental data, what do you have in mind? I have two thoughts. 
1. I can probably find an existing research, where the methodologist came up with the list that they tested on several subjects. Probably it will be possible, but most likely the data won't be for Japanese, but for English as a second language. 
2. Regarding "making the experimental data": this require a lot of people and surveys. One way to go about it is to add the new field (Natural order) to the deck and add a manual on how to reorder the deck by the "Natural order criteria" if the learner feel that it's too challenging. We can then collect the feedback to get some kind of anecdotal evidence.

**Comment by donkuri** [comment]
> I guess a google spreadsheet with the new column for the alternative order will work fine? [Something like this](https://docs.google.com/spreadsheets/d/1AE4gwmkeN-511zlIPjRFYi8xYUMvZEVGIYbBzrUk51A/edit#gid=721415458)?

Yep that would work.

> I can probably find an existing research, where the methodologist came up with the list that they tested on several subjects. Probably it will be possible, but most likely the data won't be for Japanese, but for English as a second language.

Then, how do you know the order is actually satisfactory? Unless you have a list of words that's statistically backed in some way to show it is more efficient/follows the theory, I am not sure I am willing to change the ordering of the deck. That being said, smaller changes (for instance you mentioned antonyms being put closer together) could make sense and I would consider it if you explain why they are better for learners and (ideally) provide a source backing the idea. 

Don't get me wrong, I like the idea of making the order of the deck better but I want to be sure what we're doing has a basis in facts first. Obviously, you are also free to fork the deck and change it yourself and I highly encourage doing so first anyway.

**Comment by donkuri** [comment]
Is this still being worked on?

**Comment by ganqqwerty** [comment]
we can close it since it's not going to be included in the main deck. I will release it in a fork separately
